testcafe/tests/pages/drive-model.js

```diff
@@ -85,4 +164,195 @@ export default class Page {
+  }
+
+  //@param { Array } filesIndexArray : Array of files index
+  async selectElements(filesIndexArray) {
+    console.log(`Selecting ${filesIndexArray.length} elements`)
+
+    await isExistingAndVisibile(
+      this.content_rows.nth(filesIndexArray[0]),
+      `element with index ${filesIndexArray[0]}`
+    )
+    await t.hover(this.content_rows.nth(filesIndexArray[0])) //Only one 'hover' as all checkbox should be visible once the 1st checkbox is checked
+    for (let i = 0; i < filesIndexArray.length; i++) {
+      await isExistingAndVisibile(
+        this.content_rows.nth(filesIndexArray[i]),
+        `element with index ${filesIndexArray[i]}`
+      )
+      await isExistingAndVisibile(
+        this.checboxFolderByRowIndex(filesIndexArray[i]),
+        `checkbox for element with index ${filesIndexArray[i]}`
+      )
+      await t.click(this.checboxFolderByRowIndex(filesIndexArray[i]))
+    }
+    await isExistingAndVisibile(this.cozySelectionbar, 'Cozy Selection Bar')
+  }
+
+  //@param { Array } filesIndexArray : Array of files index
+  async deleteElementsByIndex(filesIndexArray) {
+    const rowCountStart = await this.getContentRowCount('Before')
+    await this.selectElements(filesIndexArray)
+
+    await t
+      .click(this.cozySelectionbarBtnDelete)
+      .expect(this.modalDelete.visible)
+      .ok('Delete button does not show up')
+      .click(this.modalDeleteBtnDelete)
+    await t.takeScreenshot()
+    const rowCountEnd = await this.getContentRowCount('After')
+    await t.expect(rowCountEnd).eql(rowCountStart - filesIndexArray.length)
+  }
+
+  //@param {string} fileName : file name to delete
+  //this function could be improve to use an Array of filename, if needed
+  async deleteElementByName(fileName) {
+    const paragraph = this.foldersNames
+      .parent(`[class*="fil-content-row"]:not([class*="fil-content-row-head"])`)
+      .addCustomDOMProperties({
+        indexInRow: el => {
+          const nodes = Array.prototype.slice.call(el.parentElement.children)
+
+          return nodes.indexOf(el)
+        }
+      })
+
+    const index = await paragraph.withText(fileName).indexInRow
+
+    console.log(`index : ${index}`)
+    await this.deleteElementsByIndex([index])
+  }
+
+  async deleteCurrentFolder() {
+    const partialBreacrumbStart = await this.breadcrumb.child(0).innerText //We want only the 1st part of the breadcrumb to get the parent folder, so we cannot use getbreadcrumb()
+    await this.openActionMenu()
+    await t
+      .click(this.btnRemoveFolder)
+      .expect(this.modalDelete.visible)
+      .ok('Delete button does not show up')
+      .click(this.modalDeleteBtnDelete)
+    await isExistingAndVisibile(this.alertWrapper, '"successfull" modal alert')
+    await t.takeScreenshot()
+
+    await this.waitForLoading()
+
+    const breadcrumbEnd = await this.getbreadcrumb()
+    await t.expect(breadcrumbEnd).eql(partialBreacrumbStart)
+  }
 }
```

Comment:
I think you can start thinking about how managing this file in a near future and how you can divide / separate responsibilities since the file will grow again and again (not part of this PR)